src/test/java/com/github/commonsrdf/api/AbstractRDFTermFactoryTest.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.commonsrdf.api;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public abstract class AbstractRDFTermFactoryTest {
+
+	private RDFTermFactory factory;
+
+	@Test
+	public void createBlankNode() throws Exception {
+		BlankNode bnode = factory.createBlankNode();
+		String ntriplesString = bnode.ntriplesString();
+		assertTrue("ntriples must start with _:",
+				ntriplesString.startsWith("_:"));
+		assertTrue("Internal identifier can't be empty", bnode
+				.internalIdentifier().length() > 0);
+		assertEquals("ntriples does not correspond with internal identifier",
```

Comment:
The internal identifier is not the syntax for bNodes in a file.  It maybe used to generate one and an impementation may even internal use a form that is usable directly but it is not required.  The internal identifier is unique (it is in correspondence to the set of blank nodes). It may be mapped for output uses:(e.g. choose short-file-scoped "_:b0", "_:b1", ...

To quote RDF 1.1 : "IRIs, literals and **blank nodes** are distinct and distinguishable."  My emphasis.

That means it could, in the API contract, it usefully aligns to the java equality contract for RDF terms.